### PR TITLE
Restrict access to metadata form from the controller & CI automation process

### PR DESCRIPTION
**Pull request by anuveyatsu**
## CI automation

We don't update `requirements.txt` file anymore as it's unnecessary. In the CI, it uses the current code (eg, from the branch where CI is running). This is what we did to achieve this in the Dockerfile:

```bash
RUN mkdir -p $CKAN_HOME/ckanext/dcat-usmetadata
COPY . $CKAN_HOME/ckanext/dcat-usmetadata/
RUN $CKAN_HOME/bin/pip install -e $CKAN_HOME/ckanext/dcat-usmetadata
```

Next, we don't need to update template code anymore since the build files are now using static file names without hashes. This, however, has a drawback, eg, user's browser may cache the JS/CSS assets. So don't forget to disable your cache or QA the app from the incognito mode.

The JS build and moving them is now can be done by single command from project's root directory:

```
# Note this is different from running `yarn build` in metadata-app directory
yarn build
```

## Access restriction to metadata form and new tests

This PR also includes small changes to restrict access to metadata form so that only authorized users can see it.

Note, we still cannot access a new metadata form from an organization page because the template is overwritten in `ckanext-datajson` extension. We will need to open a PR to fix it:

https://github.com/GSA/ckanext-datajson/blob/master/ckanext/datajson/templates/organization/read.html#L5

### CODE REVIEW

**Comment by adborden** [comment]
@anuveyatsu can you open an issue for the organization page? Sounds like a blocker to me.

**Comment by anuveyatsu** [comment]
@adborden here is the issue https://app.zenhub.com/workspaces/datagov-multi-tenant-ckan-5cf12b601437ff1e943f0aa2/issues/gsa/datagov-ckan-multi/476

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "chance": "^1.1.6"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "build": "cd metadata-app && yarn && yarn build && cd .. && rm -r ckanext/dcat_usmetadata/public/**/* && cp -r metadata-app/build/static/* ckanext/dcat_usmetadata/public/ && cd ckanext/dcat_usmetadata/public/ && mv media static/ && mv css/main.*.chunk.css css/main.chunk.css && mv js/2.*.chunk.js js/2.chunk.js && mv js/main.*.chunk.js js/main.chunk.js && mv js/runtime-main.*.js js/runtime-main.js"
```
Comment (kmanaseryan) [inline package.json:11]: It's great that now we have single command which does all the job. Just one small thing - I would go for splitting this script into 2-3 scripts and reuse them inside the `build` script. Not sure if it makes sense, but first glance this came to my mind.

File: ckanext/dcat_usmetadata/templates/organization/read.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{% extends "organization/read_base.html" %}
```
Comment (kmanaseryan) [inline ckanext/dcat_usmetadata/templates/organization/read.html:1]: Have no idea about this new file, so just skipping to review this.

**Review by kmanaseryan — APPROVED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (anuveyatsu) [reply package.json:11]: Yes, absolutely @kmanaseryan I'll make small change here.

File: ckanext/dcat_usmetadata/templates/organization/read.html (hunk 2 above)
Comment (anuveyatsu) [reply ckanext/dcat_usmetadata/templates/organization/read.html:1]: This makes no impact on Inventory app but it suppose to change "Add dataset" button link on organization page. I've opened an issue about that here https://app.zenhub.com/workspaces/datagov-multi-tenant-ckan-5cf12b601437ff1e943f0aa2/issues/gsa/datagov-ckan-multi/476

**Comment by adborden** [comment]
@anuveyatsu how will this affect production?
> Next, we don't need to update template code anymore since the build files are now using static file names without hashes. This, however, has a drawback, eg, user's browser may cache the JS/CSS assets. So don't forget to disable your cache or QA the app from the incognito mode.

**Comment by anuveyatsu** [comment]
@adborden I have just checked response headers in the sandbox (for `main.chunk.js` file) and I don't think we need to worry about caching:

```
cache-control: max-age=0, no-cache, no-store, must-revalidate
etag: "1602075323.76-107369"
last-modified: Wed, 07 Oct 2020 12:55:23 GMT
```

As we can see from the headers, we have `etag` header for validation if content of an asset has changed as well as `cache-control` header that indicates to not cache a file.

**Pull request merged**